eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -33,6 +33,10 @@ import scala.util.Try
 object Transactions {
 
   // @formatter:off
+
+  // type alias for the map that carries around the input index of the htlc output AND the cltv of the htlc-timeout transaction
+  type HtlcTimeoutTxInputInfo = Map[Int, Long]
```

Comment:
Why not use a `Map[Int, DirectedHTlc]` here ? It could then be use to compute the HTLC success and timeout tx

Reply:
Initially i thought of this as a special case only for offered htlcs but i like the suggestion, we can get rid of 'OutputsAlreadyUsed' and store all the information (input_index, htlc) within this map. Will do.